android/expoview/build.gradle

```
@@ -273,3 +301,379 @@ dependencies {
+  if (!hermesPackagePath) {
+    throw new GradleScriptException("Could not find the hermes-engine npm package", null)
+  }
+
+  def hermesAAR = file("$hermesPackagePath/android/hermes-debug.aar")
+  if (!hermesAAR.exists()) {
+    throw new GradleScriptException("The hermes-engine npm package is missing \"android/hermes-debug.aar\"", null)
+  }
+
+  def soFiles = zipTree(hermesAAR).matching({ it.include "**/*.so" })
+
+  copy {
+    from soFiles
+    from "$reactNative/ReactAndroid/src/main/jni/first-party/hermes/Android.mk"
+    into "$thirdPartyNdkDir/hermes"
+  }
+}
+
+// Create Android.mk library module based on jsc from npm
+task prepareJSC {
+  doLast {
+    def jscPackagePath = findNodeModulePath(projectDir, "jsc-android")
+    if (!jscPackagePath) {
+      throw new GradleScriptException("Could not find the jsc-android npm package", null)
+    }
+
+    def jscDist = file("$jscPackagePath/dist")
+    if (!jscDist.exists()) {
+      throw new GradleScriptException("The jsc-android npm package is missing its \"dist\" directory", null)
+    }
+
+    def jscAAR = fileTree(jscDist).matching({ it.include "**/android-jsc/**/*.aar" }).singleFile
+    def soFiles = zipTree(jscAAR).matching({ it.include "**/*.so" })
+
+    def headerFiles = fileTree(jscDist).matching({ it.include "**/include/*.h" })
+
+    copy {
+      from(soFiles)
+      from(headerFiles)
+      from("$reactNative/ReactAndroid/src/main/jni/third-party/jsc/Android.mk")
+
+      filesMatching("**/*.h", { it.path = "JavaScriptCore/${it.name}" })
+
+      includeEmptyDirs(false)
+      into("$thirdPartyNdkDir/jsc")
+    }
+  }
+}
+
+task extractAARHeaders {
+  doLast {
+    configurations.extractHeaders.files.each {
+      def file = it.absoluteFile
+      def packageName = file.name.tokenize('-')[0]
+      copy {
+        from zipTree(file)
+        into "$reactNative/ReactAndroid/src/main/jni/first-party/$packageName/headers"
+        include "**/*.h"
+      }
+    }
+  }
+}
+
+task extractJNIFiles {
+  doLast {
+    configurations.extractJNI.files.each {
+      def file = it.absoluteFile
+      def packageName = file.name.tokenize('-')[0]
+      copy {
+        from zipTree(file)
+        into "$reactNative/ReactAndroid/src/main/jni/first-party/$packageName/"
+        include "jni/**/*"
+      }
+    }
+  }
+}
+
+task buildReanimated(dependsOn: [prepareJSC, prepareBoost, prepareDoubleConversion, prepareFolly, prepareGlog, prepareHermes, extractAARHeaders, extractJNIFiles], type: Exec) {
+  inputs.dir("src/main/JNI")
+  inputs.dir("src/main/cpp")
+  inputs.dir("src/main/Common")
+  outputs.dir("$buildDir/reanimated-ndk/all")
+  commandLine(getNdkBuildFullPath(),
+      "NDK_PROJECT_PATH=null",
+      "NDK_APPLICATION_MK=$projectDir/src/main/JNI/Application.mk",
+      "NDK_OUT=" + temporaryDir,
+      "NDK_LIBS_OUT=$buildDir/reanimated-ndk/all",
+      "THIRD_PARTY_NDK_DIR=$buildDir/third-party-ndk",
+      "JSI_PATH=$projectDir/../node_modules/react-native/ReactCommon/jsi",
```

```suggestion
      "JSI_PATH=$reactNative/ReactCommon/jsi",
```

or is this flag not used too?